File: gaia-sdk-javascript/src/graphql/request/input/CreateEdgeImpulse.ts


import VariableRegistry from "../../../api/VariableRegistry"
import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {Order} from "../enumeration/Order";
import {OrderByField} from "../enumeration/OrderByField";
import {EdgeOrderByField} from "../enumeration/EdgeOrderByField";
import {EdgeType} from "../enumeration/EdgeType";

/**
 * The specification to create an edge instance
 */
export class CreateEdgeImpulse {
public _typeName = "CreateEdgeImpulse";
    private source:Uuid;
    private target:Uuid;
    private type:String;
    private weight:Number;
    private properties:Struct;

    constructor (source:Uuid, target:Uuid, type:String, weight:Number, properties:Struct) {
        this.source = source;
        this.target = target;
        this.type = type;
        this.weight = weight;
        this.properties = properties;
    }
}

File: gaia-sdk-python/gaia_sdk/graphql/request/input/CreateEdgeImpulse.py


from typing import Callable, List
from gaia_sdk.api.VariableRegistry import VariableRegistry
from gaia_sdk.graphql.request.enumeration.Order import Order
from gaia_sdk.graphql.request.enumeration.OrderByField import OrderByField
from gaia_sdk.graphql.request.enumeration.EdgeOrderByField import EdgeOrderByField
from gaia_sdk.graphql.request.enumeration.EdgeType import EdgeType


class CreateEdgeImpulse():
    """
    The specification to create an edge instance
    """
    source: str
    target: str
    type: str
    weight: float
    properties: dict

    def __init__(self, source: str, target: str, type: str, weight: float, properties: dict):
        self.source = source
        self.target = target
        self.type = type
        self.weight = weight
        self.properties = properties

    def __eq__(self, other):
        if type(other) is type(self):
            return self.source == other.source and self.target == other.target and self.type == other.type and self.weight == other.weight and self.properties == other.properties
        return False

    def __repr__(self):
        return {'source': self.source, 'target': self.target, 'type': self.type, 'weight': self.weight, 'properties': self.properties}
